### ISSUE TITLE
None of the debug test scenario works on 4.4 cluster

### ISSUE THREAD

**Issue by amitkrout** [openshift/odo#2664]
/kind bug

## What versions of software are you using?

**Operating System:**
All supported
**Output of `odo version`:**
master
## How did you run odo exactly?
Run ```make test-cmd-debug``` on 4.4 cluster

## Actual behavior
All test scenario fails

## Expected behavior
Should work 

## Any logs, error output, etc?

Details failure log - https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_release/6892/rehearse-6892-pull-ci-openshift-odo-master-v4.4-integration-e2e-benchmark/3#1:build-log.txt%3A1186

**Comment by amitkrout** [comment]
ping @girishramnani

**Comment by amitkrout** [comment]
@girishramnani i verified successfully debug command with java and nodejs image locally against 4.4 cluster. It seems there is a problem in the test script. Will verify it further and may be send the fix.

**Comment by mik-dass** [comment]
On 4.4 even `oc` doesn't work properly. @amitkrout and me started a port-forwarding session and pinged http://localhost:5858 and it failed with the same errors as above.
```
[mrinaldas@localhost nodejs-ex]$ oc port-forward nodejs-nodejs-ex-xqbw-app-1-km76w 5858:5858
Forwarding from 127.0.0.1:5858 -> 5858
Forwarding from [::1]:5858 -> 5858
Handling connection for 5858
Handling connection for 5858
E0311 14:44:02.223700   27876 portforward.go:331] an error occurred forwarding 5858 -> 5858: error forwarding port 5858 to pod 08cbcd016c4f73bda2eb3abf900aaf1d6e94406b4b4c63da5c3f570e9cc05fee, uid : exit status 1: 2020/03/11 09:14:02 socat[206309] E connect(5, AF=2 127.0.0.1:5858, 16): Connection refused
E0311 14:44:02.971744   27876 portforward.go:331] an error occurred forwarding 5858 -> 5858: error forwarding port 5858 to pod 08cbcd016c4f73bda2eb3abf900aaf1d6e94406b4b4c63da5c3f570e9cc05fee, uid : exit status 1: 2020/03/11 09:14:02 socat[206314] E connect(5, AF=2 127.0.0.1:5858, 16): Connection refused
Handling connection for 5858
E0311 14:44:03.436736   27876 portforward.go:331] an error occurred forwarding 5858 -> 5858: error forwarding port 5858 to pod 08cbcd016c4f73bda2eb3abf900aaf1d6e94406b4b4c63da5c3f570e9cc05fee, uid : exit status 1: 2020/03/11 09:14:03 socat[206416] E connect(5, AF=2 127.0.0.1:5858, 16): Connection refused
```

**Comment by amitkrout** [comment]
@girishramnani So it seems the problem itself lies in the upstream. May we need to create an upstream issue to track it, But before that can you cross verify from your end once.

**Comment by kadel** [comment]
> @girishramnani So it seems the problem itself lies in the upstream. May we need to create an upstream issue to track it, But before that can you cross verify from your end once.

No this has nothing to do with upstream.

Checking how nodejs is run in "nodejs:latest" you can see that it is not run in "debug mode"
```
sh-4.2$ ps faxu
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
1000600+     201  0.0  0.0  11840  2912 pts/0    Ss   13:32   0:00 sh
1000600+     209  0.0  0.0  51760  3472 pts/0    R+   13:32   0:00  \_ ps faxu
1000600+       1  0.0  0.0 103032  5036 ?        Ssl  13:28   0:00 /opt/odo/bin/go-init -pre /opt/odo/bin/s2i-setup -main /opt/odo/bin/supervisord -c /opt/app-ro
1000600+      25  0.0  0.1 528248 15032 ?        Sl   13:28   0:00 /opt/odo/bin/supervisord -c /opt/app-root/conf/supervisor.conf
1000600+     168  0.1  0.4 763164 40452 ?        Sl   13:28   0:00  \_ npm
1000600+     192  0.1  0.5 602504 41084 ?        Sl   13:28   0:00      \_ node server.js
```

The reason is that nodejs:latest is not recognized as a supported component in 4.4 cluster. So it is not executed in debug mode.



```
▶ odo catalog list components
Odo Supported OpenShift Components:
NAME       PROJECT       TAGS
java       openshift     11,8,latest
nodejs     openshift     10

Odo Unsupported OpenShift Components:
NAME              PROJECT       TAGS
dotnet            openshift     2.1,3.0,3.1,latest
golang            openshift     1.11.5,latest
httpd             openshift     2.4,latest
modern-webapp     openshift     10.x,latest
nginx             openshift     1.10,1.14,latest
nodejs            openshift     12,latest
perl              openshift     5.26,latest
php               openshift     7.2,7.3,latest
python            openshift     2.7,3.6,latest
ruby              openshift     2.4,2.5,latest
```

**Comment by kadel** [comment]
@girishramnani  @amitkrout this has the highest priority, where we are with this?

**Comment by girishramnani** [comment]
@amitkrout This seems to resolved by https://github.com/openshift/odo/pull/2729, can you please confirm?

**Comment by amitkrout** [comment]
> @amitkrout This seems to resolved by #2729, can you please confirm?

Yes, you are right. I have test the your pr #2729 against 4.4 cluster for debug test.
